lib/stanford-mods/name.rb

```diff
@@ -70,18 +64,23 @@ def non_collector_person_authors
       def collectors_w_dates
         result = []
         @mods_ng_xml.personal_name.each do |n|
-          unless n.role.size == 0
-            n.role.each { |r|
-              if (r.authority.include?('marcrelator') && r.value.include?('Collector')) ||
-                  r.roleTerm.valueURI.first == COLLECTOR_ROLE_URI
-                result << n.display_value_w_date
-              end
-            }
-          end
+          next if n.role.size.zero?
+          n.role.each { |r|
+            result << n.display_value_w_date if includes_marc_relator_collector_role?(r)
+          }
         end
         result unless result.empty?
       end
 
+      COLLECTOR_ROLE_URI = 'http://id.loc.gov/vocabulary/relators/col'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.